Sources/CryptoSwift/SHA2.swift

```diff
@@ -273,8 +273,13 @@ final class SHA2 : HashProtocol {
         result.reserveCapacity(hh.count / 4)
         variant.resultingArray(hh).forEach {
             let item = $0.bigEndian
-            result += [UInt8(item & 0xff), UInt8((item >> 8) & 0xff), UInt8((item >> 16) & 0xff), UInt8((item >> 24) & 0xff),
-                       UInt8((item >> 32) & 0xff),UInt8((item >> 40) & 0xff), UInt8((item >> 48) & 0xff), UInt8((item >> 56) & 0xff)]
+            var chunk = [UInt8]()
+            chunk.reserveCapacity(8)
```

Comment:
no, but you can create array with default values ;) nvm.

[UInt8](count: 8, repeatedValue: 0)  

--  
Marcin


On Thursday 18 February 2016 at 13:54, Oscar De Moya wrote:

> In Sources/CryptoSwift/SHA2.swift (https://github.com/krzyzanowskim/CryptoSwift/pull/215#discussion_r53309473):
> > @@ -273,8 +273,13 @@ final class SHA2 : HashProtocol { > result.reserveCapacity(hh.count / 4) > variant.resultingArray(hh).forEach { > let item = $0.bigEndian > - result += [UInt8(item & 0xff), UInt8((item >> 8) & 0xff), UInt8((item >> 16) & 0xff), UInt8((item >> 24) & 0xff), > - UInt8((item >> 32) & 0xff),UInt8((item >> 40) & 0xff), UInt8((item >> 48) & 0xff), UInt8((item >> 56) & 0xff)] > + var chunk = [UInt8]() > + chunk.reserveCapacity(8)  
>  
> I couldn't use the square brackets notation chunk[i] = ... there. I got an "Index out of range" error. It seems we can't create an Array that has pre-allocated memory but does not contain elements: http://stackoverflow.com/questions/24427905/array-with-capacity-in-swift/24428086#24428086
> with the reserveCapacity call append should behave better, though.
>  
> —
> Reply to this email directly or view it on GitHub (https://github.com/krzyzanowskim/CryptoSwift/pull/215/files#r53309473).